samples/dnn/text_detection.cpp

```diff
@@ -159,3 +215,16 @@ void decode(const Mat& scores, const Mat& geometry, float scoreThresh,
         }
     }
 }
+
+void decode_text(Mat& scores, std::string& text) {
```

Comment:
The decode_text here is wrong all the adjacent same characters should be removed before removing the background classes ('-')